### PR TITLE
Product listing

### PR DESCRIPTION
**Pull request by BALAJI141031**
what to check:
1. prodcts route  file and application of filters

### CODE REVIEW

File: src/components/cards/index.css [hunk 1]
```diff
@@ -0,0 +1,82 @@
+
+/* .overlay-card{
```
Comment (dipen2000) [inline src/components/cards/index.css:2]: remove the commented code , if that is there just for your reference for right now then it is good.

File: src/components/cards/serviceCard.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import './index.css'
+const ServiceCard=()=>{
+    return <div className="serviceCard flex-V-center-VH"><div class="sq-card">
+    <img src="https://b.zmtcdn.com/webFrontend/e5b8785c257af2a7f354f1addaf37e4e1647364814.jpeg?fit=around|402:360&crop=402:360;*,*" alt="service-card" />
```
Comment (dipen2000) [inline src/components/cards/serviceCard.js]: you should wrap this img tag with a img container div and make your image responsive by adding a img-resp class on that have the img tag be width : 100% and height : 100% and then define the height and width on the wrapper div and add those object-fit and stuff to the image

File: src/components/index.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import {CardWithText} from './cards/cardWithText'
+import {ProductCard} from './cards/productCard'
+import {ServiceCard} from './cards/serviceCard'
+import {TextCard} from './cards/textCard'
+import {TextOverlayCard} from './cards/textOverlay'
+import {WishlistCard} from './cards/wishlistCard'
```
Comment (dipen2000) [inline src/components/index.js]: use barrel import here.

File: src/contexts/filterContext.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            case "delivery":
+                    return {...filterState,delivery:filterAction.payload}
+            case "rating":
+                    return {...filterState,rating:filterAction.payload}
+            default:
+                return {}
+        }
+    }
+
+    const [{sortBy,productsList,vegetarian,delivery,rating},dispatchFilter]=useReducer(reducerFn,{allProducts:true,sortBy:null,productsList:[],vegetarian:true,delivery:false,rating:false})
```
Comment (dipen2000) [inline src/contexts/filterContext.js:25]: You should store the initial state value a variable and then use it here for clean code and readability purposes.

File: src/contexts/filterContext.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    
+    // filter functions        
+    const sortProducts=(products,sortBy)=>{
+        console.log(sortBy,"check1",products)
+        if(sortBy==="low_to_high"){
+            return products.sort((a,b)=>a.price-b.price)
+        }else if (sortBy==="high_to_low"){ 
+            return products.sort((a,b)=>b.price-a.price)
+        }else{    
+            return products
```
Comment (dipen2000) [inline src/contexts/filterContext.js:36]: Use switch for this.

File: src/providers/index.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import { BrowserRouter as Router } from 'react-router-dom';
+import {Sidebar} from '../contexts/sidebarContext'
+import {FilterProvider} from '../contexts/filterContext'
+
+const Providers=({children})=>{
+    return <Router>
+    <FilterProvider>
+        <Sidebar>{children}</Sidebar>
+    </FilterProvider>
+    </Router>
```
Comment (dipen2000) [inline src/providers/index.js]: Use something like prettier to auto-format this code.

**Review by dipen2000 — APPROVED** [review]
Just look over some suggestions that i mentioned, otherwise good to go.

File: src/providers/index.js (hunk 6 above)
Comment (BALAJI141031) [reply src/providers/index.js]: ok

File: src/contexts/filterContext.js (hunk 5 above)
Comment (BALAJI141031) [reply src/contexts/filterContext.js:36]: ok

**Pull request merged**